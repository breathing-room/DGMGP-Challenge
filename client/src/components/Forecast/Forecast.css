.container {
    width: 100px;
    display: flex;
    flex-direction: column;
    padding: 18px;
    background-color: rgba(3, 190, 190, 0.192);
    /* background-color: rgb(222, 253, 253); */
    box-shadow: 2px 2px 6px -2px rgba(0,0,0,0.46);
    border-radius: 2px;
    justify-content: center;
    max-height: 175px;
    border-bottom: 2px solid rgba(253, 116, 36, 0.521);

}

.main-content {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
}

.main-content h3 {
    font-weight: normal;
}

.main-content h4 {
    margin: 0 auto;
    font-size: 22px;
    font-weight: 500
}

.weather-icon {
    /* padding: 5px; */
    margin-bottom: 10px;
}

.description-container {
    margin: 0 auto;
    font-size: 15px;
}

.description {
    font-size: 13px;
    font-weight: 500;
    font-style: italic;
    padding: 5px;
}