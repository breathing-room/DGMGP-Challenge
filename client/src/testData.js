// Testing data grabbed from actual call to Open Weather API
const testData = {
	"list": 
		[
			{
				"dt": 1585677600,
				"sunrise": 1585655402,
				"sunset": 1585700293,
				"temp": {
					"day": 78.96,
					"min": 57.4,
					"max": 80.26,
					"night": 57.4,
					"eve": 68.56,
					"morn": 77.47
				},
				"feels_like": {
					"day": 72.03,
					"night": 48.29,
					"eve": 59.45,
					"morn": 76.19
				},
				"pressure": 1011,
				"humidity": 61,
				"weather": [
					{
						"id": 501,
						"main": "Rain",
						"description": "moderate rain",
						"icon": "10d"
					}
				],
				"speed": 21.21,
				"deg": 316,
				"clouds": 46,
				"rain": 3.51
			},
			{
				"dt": 1585764000,
				"sunrise": 1585741730,
				"sunset": 1585786728,
				"temp": {
					"day": 68.85,
					"min": 54.75,
					"max": 71.89,
					"night": 63.1,
					"eve": 67.75,
					"morn": 54.75
				},
				"feels_like": {
					"day": 63.81,
					"night": 61.74,
					"eve": 63.73,
					"morn": 47.53
				},
				"pressure": 1018,
				"humidity": 38,
				"weather": [
					{
						"id": 800,
						"main": "Clear",
						"description": "sky is clear",
						"icon": "01d"
					}
				],
				"speed": 5.77,
				"deg": 39,
				"clouds": 6
			},
			{
				"dt": 1585850400,
				"sunrise": 1585828058,
				"sunset": 1585873164,
				"temp": {
					"day": 75.24,
					"min": 61,
					"max": 75.42,
					"night": 65.89,
					"eve": 70.09,
					"morn": 61
				},
				"feels_like": {
					"day": 70.36,
					"night": 63.93,
					"eve": 67.01,
					"morn": 57.06
				},
				"pressure": 1018,
				"humidity": 45,
				"weather": [
					{
						"id": 803,
						"main": "Clouds",
						"description": "broken clouds",
						"icon": "04d"
					}
				],
				"speed": 10,
				"deg": 100,
				"clouds": 72
			},
			{
				"dt": 1585936800,
				"sunrise": 1585914386,
				"sunset": 1585959599,
				"temp": {
					"day": 78.55,
					"min": 63.54,
					"max": 80.28,
					"night": 69.21,
					"eve": 73.13,
					"morn": 63.54
				},
				"feels_like": {
					"day": 74.03,
					"night": 67.51,
					"eve": 69.3,
					"morn": 60.94
				},
				"pressure": 1017,
				"humidity": 36,
				"weather": [
					{
						"id": 803,
						"main": "Clouds",
						"description": "broken clouds",
						"icon": "04d"
					}
				],
				"speed": 7.85,
				"deg": 122,
				"clouds": 67
			},
			{
				"dt": 1586023200,
				"sunrise": 1586000715,
				"sunset": 1586046035,
				"temp": {
					"day": 73.24,
					"min": 65.5,
					"max": 73.69,
					"night": 66.99,
					"eve": 65.5,
					"morn": 68.34
				},
				"feels_like": {
					"day": 71.2,
					"night": 66.92,
					"eve": 62.26,
					"morn": 68.85
				},
				"pressure": 1017,
				"humidity": 63,
				"weather": [
					{
						"id": 501,
						"main": "Rain",
						"description": "moderate rain",
						"icon": "10d"
					}
				],
				"speed": 9.33,
				"deg": 73,
				"clouds": 100,
				"rain": 3.48
			}
		]
}

export default testData;