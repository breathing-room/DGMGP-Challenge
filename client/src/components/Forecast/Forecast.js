import React from 'react';
import PropTypes from 'prop-types'
import moment from 'moment';
import './Forecast.css'

const Forecast = ({ main, date, description }) => {
    const desc = description[0].main.toLowerCase();
    // Retrieve weather icon from Open Weather Map API
    const iconUrl = `https://openweathermap.org/img/wn/${description[0].icon}.png`
    // Convert Unix date into day of the week
    const day = moment.unix(date).format('dddd');
    return (
        <div className='container'> 
            <div className='main-content'>
                <h3>{day}</h3>
                <div className='weather-icon'>
                    <img src={iconUrl} alt='Weather Icon'/>
                </div>
                <h4>{main}Â° F</h4>
                <p className='description'>{desc}</p>
            </div>
        </div>
    )
}

Forecast.propTypes = {
    main: PropTypes.number,
    date: PropTypes.number,
    description: PropTypes.array,
}

export default Forecast;