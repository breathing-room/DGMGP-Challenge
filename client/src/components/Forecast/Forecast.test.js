import React from 'react';
import Forecast from './Forecast'
import testData from '../../testData'
import { cleanup } from '@testing-library/react';
import { mount } from 'enzyme'

describe('Forecast component', () => {
  let wrapper;
  const firstDayData = testData.list[0]
  const main = firstDayData.temp.day;
  const description = firstDayData.weather;
  const date = firstDayData.dt;
  
  beforeEach(() => {
    wrapper = mount(
      <Forecast main={main} description={description} date={date}/>
    );
  })

  afterEach(cleanup)

  test('it renders correctly', () => {  
      expect(wrapper).toMatchSnapshot();
  });

})

