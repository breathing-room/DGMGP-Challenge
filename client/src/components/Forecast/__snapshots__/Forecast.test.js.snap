// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast component it renders correctly 1`] = `ReactWrapper {}`;
